src/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -50,20 +50,34 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
                 }
 
                 msbuildSdksDir = Path.Combine(netcoreSdkDir, "Sdks");
-                netcoreSdkVersion = new DirectoryInfo(netcoreSdkDir).Name;;
-            }
+                netcoreSdkVersion = new DirectoryInfo(netcoreSdkDir).Name;
 
-            if (!IsNetCoreSDKOveridden(netcoreSdkVersion) && 
-                IsNetCoreSDKSmallerThanTheMinimumVersion(netcoreSdkVersion, sdkReference.MinimumVersion))
-            {
-                return factory.IndicateFailure(
-                    new[]
-                    {
-                        $"Version {netcoreSdkVersion} of the SDK is smaller than the minimum version"
-                        + $" {sdkReference.MinimumVersion} requested. Check that a recent enough .NET Core SDK is"
-                        + " installed, increase the minimum version specified in the project, or increase"
-                        + " the version specified in global.json."
-                    });
+                if (IsNetCoreSDKSmallerThanTheMinimumVersion(netcoreSdkVersion, sdkReference.MinimumVersion))
+                {
+                    return factory.IndicateFailure(
+                        new[]
+                        {
+                            $"Version {netcoreSdkVersion} of the SDK is smaller than the minimum version"
+                            + $" {sdkReference.MinimumVersion} requested. Check that a recent enough .NET Core SDK is"
+                            + " installed, increase the minimum version specified in the project, or increase"
+                            + " the version specified in global.json."
+                        });
+                }
+
+                var minimumMSBuildVersionString =
+                    File.ReadAllLines(Path.Combine(netcoreSdkDir, "minimumMSBuildVersion"))[0];
+                var minimumMSBuildVersion = Version.Parse(minimumMSBuildVersionString);
+                if (context.MSBuildVersion.CompareTo(minimumMSBuildVersion) == -1)
```

Comment:
The CompareTo contract is less-than-zero for less-than not equal-to-negative-one. This is effectively taking a dependency on an implementation detail of Version.CompareTo.

We can use `operator <` to make the code easier to read.

We should also change the Compare of FXVersions to check < 0. That one is less bad because the implementation we're depending on is ours, but still Compare methods should be interpreted as <0, 0, or >0 not -1,0,1.